ral-facilities/datagateway
# Adding ability to use DatePicker or DateTimePicker in date filter column #1175

## Description
Depending on if a `filterByTime` prop is passed, the date filter component will return either a `KeyboardDatePicker` or a `KeyboardDateTimePicker`. Thus, the admin download table now contains a `KeyboardDateTimePicker` for more accurate filtering of results, as per user request.

This required some amendments to how time is passed to the backend date query. Previously, if a date was passed, it was assumed the time value would be 00:00:00 or 23:59:59, depending on `startDate` or `endDate` being changed. Now we must provide a time value if the user specifies one.

Fortunately, the formatting for this is done in the `dateColumnFilter` component. However, the GUI for `KeyboardDateTimePicker` can only be accurate to hours and minutes - we can't enter a value for seconds. 

This causes a slight amendment to queries - if we enter an end date of `2022-04-06 23:59`, we won't get any results for `2022-04-06 23:59:00` < x < `2022-04-06 23:59:59`. We'd need to update the query to an end date of `2022-04-07 00:00`. If anyone has any thoughts on this, how this might impact users, other unintended side-effects, solutions for including a value for seconds, etc. please comment!

## Testing instructions
Because this is a change to the globally-used date filter, it is important to test date filtering in a variety of pages, including dataview, search, download and admin download. Currently, only the admin download table wants date and time filtering so make sure that other filters are date filter only and continue to work as expected. Finally, check to make sure that date and time values can be filtered on the admin download table correctly.


## Agile board tracking
Closes #1175
